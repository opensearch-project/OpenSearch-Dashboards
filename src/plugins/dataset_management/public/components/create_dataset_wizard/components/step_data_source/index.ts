/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export { StepDataSource } from './step_data_source';
