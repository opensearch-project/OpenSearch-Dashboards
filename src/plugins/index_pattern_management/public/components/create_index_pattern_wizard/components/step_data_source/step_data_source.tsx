/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiPageContent } from '@elastic/eui';
import React, { useState } from 'react';
import { DataSourceRef } from 'src/plugins/index_pattern_management/public/types';
import { StepInfo } from '../../types';

import { Header } from './components/header';

interface StepDataSourceProps {
  goToNextStep: (dataSourceRef: DataSourceRef) => void;
  stepInfo: StepInfo;
}

export const StepDataSource = (props: StepDataSourceProps) => {
  const { goToNextStep, stepInfo } = props;

  const [selectedDataSource, setSelectedDataSource] = useState<DataSourceRef>();
  const [isNextStepDisabled, setIsNextStepDisabled] = useState(true);

  const onDataSourceSelected = (id: string, selectedType: string, title: string) => {
    const selected = { id, type: selectedType, title };

    setSelectedDataSource(selected);
    setIsNextStepDisabled(false);
  };

  const renderContent = () => {
    return (
      <EuiPageContent>
        <Header
          onDataSourceSelected={onDataSourceSelected}
          dataSourceRef={selectedDataSource!}
          goToNextStep={() => goToNextStep(selectedDataSource!)}
          isNextStepDisabled={isNextStepDisabled}
          stepInfo={stepInfo}
        />
      </EuiPageContent>
    );
  };

  return <>{renderContent()}</>;
};
