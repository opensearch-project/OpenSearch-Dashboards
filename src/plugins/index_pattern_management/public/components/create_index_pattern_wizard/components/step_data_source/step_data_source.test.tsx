/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { shallow } from 'enzyme';
import { StepDataSource } from './step_data_source';

describe('StepDataSource', () => {
  it('should render normally', () => {
    const component = shallow(
      <StepDataSource
        goToNextStep={() => {}}
        stepInfo={{ totalStepNumber: 0, currentStepNumber: 0 }}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
