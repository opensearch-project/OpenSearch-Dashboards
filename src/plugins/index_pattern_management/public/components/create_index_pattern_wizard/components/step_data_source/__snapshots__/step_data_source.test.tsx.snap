// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepDataSource should render normally 1`] = `
<Fragment>
  <EuiPageContent>
    <Header
      goToNextStep={[Function]}
      isNextStepDisabled={true}
      onDataSourceSelected={[Function]}
      stepInfo={
        Object {
          "currentStepNumber": 0,
          "totalStepNumber": 0,
        }
      }
    />
  </EuiPageContent>
</Fragment>
`;
